upstairs/src/lib.rs

```diff
@@ -877,6 +983,21 @@ impl Upstairs {
          */
         assert_eq!(opt.target.len(), 3);
 
+        // create an encryption context if a key is supplied.
+        let encryption_context = opt.key_bytes().map(|key| {
+            UpstairsEncryptionContext::new(
+                key,
+                /*
+                 * XXX: It would be good to do BlockOp::QueryBlockSize here, but
```

Comment:
I guess, yeah, this has to be an XXX now.  The block size is not available to up_main until the downstairs have all come online and are receiving IO.  We will have to see if the block size can be passed from Nexus to propolis to crucible.  Otherwise we have to wait to setup encryption till we have received the info from downstairs (Which may mean some other state transition stuff) and have the upstairs only allow query commands until the encryption is setup.